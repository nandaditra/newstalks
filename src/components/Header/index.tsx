import { Link, useNavigate } from "react-router-dom"

const Header= () => {
   const navigate = useNavigate();
   const handleSearch = async(event:React.FormEvent<HTMLFormElement>) => {
       event.preventDefault();
       const target = await event.target as typeof event.target & {
         search: { value: string };
      };
      
       const keyword = target.search.value
       if(keyword) {
          navigate(`/search/${keyword}`)
       }
   }

   return (
     <header className="sticky-top bg-white">
        <nav className="container navbar navbar-expand-lg">
            <div className="container-fluid">
                <Link className="mr-0 navbar-brand fw-bold" to={"/"}>
                   <span className="text-danger">NEWS</span>
                   <span className="text-primary">TALK</span>
                </Link>
                <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>
                <div className="collapse navbar-collapse" id="navbarSupportedContent">
                  <ul className="navbar-nav ms-auto mb-2 mb-lg-0">
                       <li className="nav-item">
                            <Link className="nav-link active" to={"/"}>Trending</Link>
                        </li>
                        <li className="nav-item">
                            <Link className="nav-link" to={"/news"}>News</Link>
                        </li>
                        <li className="nav-item">
                            <Link className="nav-link" to={"/selebrities"} >Selebrities</Link>
                        </li>
                        <li className="nav-item">
                            <Link className="nav-link" to={"/sports"} >Sports</Link>
                        </li>
                        <li className="nav-item">
                            <Link className="nav-link" to={"/technology"} >Technology</Link>
                        </li>
                        <form className="d-flex" role="search" onSubmit={handleSearch}>
                            <input className="form-control me-2" type="search" placeholder="Search" aria-label="Search" name="search"/>
                            <button className="btn btn-primary" type="submit">Search</button>
                        </form>
                  </ul>
                </div>
            </div>
            </nav>
     </header>
   )
}

export default Header