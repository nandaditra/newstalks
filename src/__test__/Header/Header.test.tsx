import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer';
import Header from '../../layout/Header';

test('renders the "NEWS" message', () => {
  render(<Header />);
  const newsText = screen.getByText("NEWS");
  expect(newsText).toBeInTheDocument();
});

test('renders the "TALK" message', () => {
  render(<Header />);
  const newsText = screen.getByText("TALK");
  expect(newsText).toBeInTheDocument();
});

it('performs snapshot testing', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
});