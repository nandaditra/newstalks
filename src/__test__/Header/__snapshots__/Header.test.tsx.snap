// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`performs snapshot testing 1`] = `
<header
  className="sticky-top bg-white"
>
  <nav
    className="container navbar navbar-expand-lg"
  >
    <div
      className="container-fluid"
    >
      <a
        className="mr-0 navbar-brand fw-bold"
        href="/"
      >
        <span
          className="text-danger"
        >
          NEWS
        </span>
        <span
          className="text-primary"
        >
          TALK
        </span>
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav ms-auto mb-2 mb-lg-0"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link active"
              href="/"
            >
              Trending
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/news"
            >
              News
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/selebrities"
            >
              Selebrities
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/sports"
            >
              Sports
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/technology"
            >
              Technology
            </a>
          </li>
          <form
            className="d-flex"
            onSubmit={[Function]}
            role="search"
          >
            <input
              aria-label="Search"
              className="form-control me-2"
              name="search"
              placeholder="Search"
              type="search"
            />
            <button
              className="btn btn-primary"
              type="submit"
            >
              Search
            </button>
          </form>
        </ul>
      </div>
    </div>
  </nav>
</header>
`;
